rosbag2_test_common/include/rosbag2_test_common/publisher_manager.hpp

```diff
@@ -75,13 +75,16 @@ class PublisherManager
 
   template<class T>
   void add_publisher(
-    const std::string & topic_name, std::shared_ptr<T> message, size_t expected_messages = 0)
+    const std::string & topic_name,
+    std::shared_ptr<T> message,
+    size_t expected_messages = 0,
+    const rclcpp::QoS & qos = rclcpp::QoS{10})
```

Comment:
Use `SystemDefaultsQoS`?

Reply:
SystemDefault != Default and should never be used, it has undefined behavior because it does not define a default but rather shops the value out the specific rmw implementation.

Reply:
But you're right that we can't just have the 10 constant sitting in here. Best to initialize with `rmw_qos_profile_default`

Reply:
Apparently `SystemDefaultsQoS` doesn't guarantee consistency across tests since each RMW implementation has it's own definition of System Default.

Reply:
It looks like the [QoS constructor](https://github.com/ros2/rclcpp/blob/01a6befdde7f78b6f7788e6180cc5ed38390e955/rclcpp/include/rclcpp/qos.hpp#L60) uses `rmw_qos_profile_default` by default. It still needs a `QoSInitialization` so maybe something along the lines of:

```C++
const size_t DEFAULT_HISTORY = 10;
...
const rclcpp::QoS & qos = rclcpp::QoS{KeepLast(DEFAULT_HISTORY)};
```

Since the default in [`qos_profiles.h`](https://github.com/ros2/rmw/blob/2f954d77a102042fa5c13540e2db300d549d7ad3/rmw/include/rmw/qos_profiles.h#L54) is 10.

Reply:
Update: Using instead

```c++
const rclcpp::QoS & qos = rclcpp::QoS{KeepAll()};
```